models/fetchAllArticles.model.js

```diff
@@ -1,31 +1,43 @@
 const db = require("../db/connection.js");
 
-function fetchAllArticles(sort_by = "created_at", order = "DESC") {
+function fetchAllArticles(sort_by = "created_at", order = "DESC", topic) {
   const validCols = ["comment_count", "created_at", "article_id", "title", "votes", "author", "topic"];
   const validOrder = ["ASC", "DESC"];
 
   if (!validCols.includes(sort_by)) sort_by = "created_at";
   if (!validOrder.includes(order)) order = "DESC";
 
-  return db
-    .query(
-      `SELECT
-        articles.author,
-        articles.title,
-        articles.article_id,
-        articles.topic,
-        articles.created_at,
-        articles.votes,
-        articles.article_img_url,
-        COUNT(comments.comment_id) ::INT AS comment_count
-      FROM comments
-      RIGHT JOIN articles ON comments.article_id = articles.article_id
-      GROUP BY articles.article_id
-      ORDER BY ${sort_by} ${order}`
-    )
-    .then(({ rows }) => {
-      return rows;
-    });
+  const queryVals = [];
+
+  let queryStr = `
+  SELECT
+    articles.author,
+    articles.title,
+    articles.article_id,
+    articles.topic,
+    articles.created_at,
+    articles.votes,
+    articles.article_img_url,
+  COUNT(comments.comment_id) ::INT AS comment_count
+  FROM comments
+  RIGHT JOIN articles ON comments.article_id = articles.article_id
+  `;
+
+  if (topic) {
+    queryStr += ` WHERE articles.topic = $1`;
+    queryVals.push(topic);
+  }
+
+  queryStr += `
+    GROUP BY articles.article_id
+    ORDER BY ${sort_by} ${order}
+  `;
+  return db.query(queryStr, queryVals).then(({ rows }) => {
+    if (rows.length === 0 && topic) {
```

Comment:
what if the topic is found and it doesn't have any articles like the topic 'paper'